File: wallet/src/main/java/com/worldventures/wallet/domain/converter/ApiPlaceToWalletPlaceConverter.java
package com.worldventures.wallet.domain.converter;

import com.worldventures.core.converter.Converter;
import com.worldventures.wallet.domain.entity.lostcard.ImmutableWalletPlace;
import com.worldventures.wallet.domain.entity.lostcard.WalletPlace;
import com.worldventures.wallet.service.lostcard.command.http.model.ApiPlace;

import io.techery.mappery.MapperyContext;

public class ApiPlaceToWalletPlaceConverter implements Converter<ApiPlace, WalletPlace> {

   @Override
   public WalletPlace convert(MapperyContext mapperyContext, ApiPlace apiPlace) {
      return ImmutableWalletPlace.builder().name(apiPlace.name()).build();
   }

   @Override
   public Class<ApiPlace> sourceClass() {
      return ApiPlace.class;
   }

   @Override
   public Class<WalletPlace> targetClass() {
      return WalletPlace.class;
   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/converter/ApiPlaceToWalletPlaceConverter.kt
package com.worldventures.wallet.domain.converter

import com.worldventures.core.converter.Converter
import com.worldventures.wallet.domain.entity.lostcard.WalletPlace
import com.worldventures.wallet.service.lostcard.command.http.model.ApiPlace

import io.techery.mappery.MapperyContext

class ApiPlaceToWalletPlaceConverter : Converter<ApiPlace, WalletPlace> {

   override fun convert(context: MapperyContext, source: ApiPlace): WalletPlace = WalletPlace(source.name)

   override fun sourceClass(): Class<ApiPlace> = ApiPlace::class.java

   override fun targetClass(): Class<WalletPlace> = WalletPlace::class.java
}
